game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/BattleSteps.java

```diff
@@ -31,8 +37,13 @@
   final @NonNull GamePlayer defender;
   final @NonNull Collection<Unit> offensiveAa;
   final @NonNull Collection<Unit> defendingAa;
+
+  @Getter(onMethod = @__({@Override}))
   final @NonNull Collection<Unit> attackingUnits;
```

Comment:
Attacking and defending units being together pretty much always looks like the [data clump](https://en.wikipedia.org/wiki/Data_clump) code smell. It appears that `BattleState` could instead become a first class object and be a relatively simple value object. 

It might make sense to rename `BattleState` to `UnitsInBattle` or `BattleUnits`, or `BattleParticipants` as we are not capturing the full battle state but just the units in battle.


If serialization were not a concern, then we could replace both attacking and defending units with a `BattleState`. To fix the data clump code smell we could:
- add a transient `BattleState` object. Whenever we assign a battle state object then we write the corresponding values to attacking and defending units. This way we would get serialization and any usages of the data would be through battle state. It might be possible as well to override the serialization write-object method and to serialize attacking units and defending units through a battle state. Serialization proxy might be a way to do that, though the proxy itself could break compatibility alone.

Fundamentally `BattleState` is not really an interface, it's more data that can be present in an object and I suspect it makes sense to simply have a reference to it.

WDYT?

Reply:
I agree.  It's what I'd like to get to.  Once the `BattleState` interface has been fully built, I'll then know what is all the parts of `MustFightBattle` that is part of the state and I'd like to make a value object that holds that data.  I'd also like to convert the attacking/defending units into `FiringGroups` (from my older PR that I closed).  Then it doesn't need separate attacking/defending units, attacking/defending aa, and bombarding units.